### DISCUSSION TITLE
Event when server starts listening

### DISCUSSION THREAD

**Discussion by rightaway** [koajs/koa#1551] [Q&A] [answered]
In a koa app I'd like to be able to do something when the server's `listen()` completes and it is ready to accept connections. This is what I have

```js
app.listen().on("error", errorCallback)
```

Is there a `ready` event like this?

```js
app.listen().on("error", errorCallback).on("ready", readyCallback)
```

**Comment by masx200** [comment, marked as answer]
on("listening", readyCallback)
